lib/server.js

```diff
@@ -108,19 +108,25 @@ exports.create = function create(createOptions) {
 
     try {
       if (this.useRouter) {
-        match(thing, function (error, redirectLocation, renderProps) {
+        match({ routes:createOptions.routes, location:thing},
+          function(error, redirectLocation, renderProps) {
           if (redirectLocation) {
+            debug('server.js match 302 %s', redirectLocation);
```

Comment:
@samsel, just thinking out of mind, finding some synergy

i see that in **match.js**, there are only 3 status code we need to handle 302, 404, 500

for `404` & `500`, we can allow developer to pass 2 react components in **config.json** and if it's not supplied, we provide them with default ones.

for `302`, i am not sure what is the expectation for react-engine.

* using the logic `name[0]==='/'` to redirect back to match
  * assumption here is that match will not do nested matching (yet to confirm)

OR

* see if it matches any standalone views outside the route and respond with that component view

[Reference] -  callback for redirectLocation in [React-Router](https://github.com/rackt/react-router/blob/master/modules/useRoutes.js#L73).